dev-client/src/components/sheets/FormOverlaySheet.tsx

```diff
@@ -25,100 +25,53 @@ import {
 import {useFocusEffect} from '@react-navigation/native';
 
 import {BackdropComponent} from 'terraso-mobile-client/components/BackdropComponent';
-import {BigCloseButton} from 'terraso-mobile-client/components/buttons/icons/common/BigCloseButton';
 import {
   ModalContext,
   ModalHandle,
-  ModalProps,
+  ModalTrigger,
 } from 'terraso-mobile-client/components/modals/Modal';
-import {
-  Box,
-  Column,
-  Row,
-} from 'terraso-mobile-client/components/NativeBaseAdapters';
+import {FormOverlaySheetHeader} from 'terraso-mobile-client/components/sheets/FormOverlaySheetHeader';
+import {useGorhomModalHandleRef} from 'terraso-mobile-client/components/sheets/hooks/gorhomHooks';
 import {useHeaderHeight} from 'terraso-mobile-client/hooks/useHeaderHeight';
 
-type Props = ModalProps & {
-  fullHeight?: boolean;
-  maxHeight?: number;
-  scrollable?: boolean;
-};
+export type FormOverlaySheetProps = React.PropsWithChildren<{
+  trigger?: ModalTrigger;
+}>;
 
-/**
- * To be simplified internally with FormOverlaySheet component work (mobile-client ticket #1774).
+/*
+ * Half-screen overlay sheet with a top header labeled "done", used for
+ * presenting form controls to the user.
  */
-export const FormOverlaySheet = forwardRef<
-  ModalHandle,
-  React.PropsWithChildren<Props>
->(
-  (
-    {
-      Header,
-      children,
-      trigger,
-      Closer,
-      fullHeight = false,
-      scrollable = true,
-      maxHeight,
-    },
-    forwardedRef,
-  ) => {
+export const FormOverlaySheet = forwardRef<ModalHandle, FormOverlaySheetProps>(
+  ({children, trigger}: FormOverlaySheetProps, ref) => {
     const {headerHeight} = useHeaderHeight();
-
-    const ref = useRef<GorhomBottomSheetModal>(null);
-    const methods = useMemo(
-      () => ({
-        onClose: () => ref.current?.dismiss(),
-        onOpen: () => ref.current?.present(),
-      }),
-      [ref],
-    );
-    useImperativeHandle(forwardedRef, () => methods, [methods]);
-
-    const contents =
-      Header || Closer ? (
-        <Column padding="md">
-          <Row alignItems="center" mb="md">
-            {Header}
-            <Box flex={1} />
-            {Closer === undefined ? (
-              <BigCloseButton onPress={methods.onClose} />
-            ) : (
-              Closer
-            )}
-          </Row>
-          {children}
-        </Column>
-      ) : (
-        children
-      );
+    const {modalRef, methods} = useGorhomModalHandleRef(ref);
 
     return (
       <>
-        {trigger && (
-          <Pressable onPress={methods.onOpen}>
-            {trigger(methods.onOpen)}
-          </Pressable>
-        )}
+        {trigger && trigger(methods.onOpen)}
```

Comment:
The indirection with trigger and refs feels challenging to read -- it's not a result of your change, but wondering if there is some commenting here that could make it faster for a reader to understand

Reply:
I'll see what I can do, I agree that it's confusing.